src/utils/SocketClient.hxx

```diff
@@ -254,34 +259,100 @@ private:
 
             if (ai_ret == 0 && addr_)
             {
-                /* able to resolve the hostname */
-                fd_ = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
-                if (fd_ >= 0)
+                /* able to resolve the hostname to an address */
+                bool addr_okay = true;
+                if (addr_->ai_addr->sa_family != AF_INET)
```

Comment:
what happened previously when you resolved the target host to an ipv6 address?

Reply:
Didn't work.  I suspect we want IPv6 support, but we don't have it today.